sensors/sqs_service_sensor.py

```diff
@@ -0,0 +1,155 @@
+        config = self.config
+        if prefix:
+            config = self._config.get(prefix, {})
+
+        value = self._sensor_service.get_value('aws.%s' % (key), local=False)
+        if not value:
+            value = config.get(key, None)
+
+        if not value and config.get('setup', None):
+            value = config['setup'].get(key, None)
+
+        return value
+
+    def _may_setup_sqs(self):
+        queues = self._get_config_entry(key='input_queues', prefix='sqs_sensor')
+
+        # XXX: This is a hack as from datastore we can only receive a string while
+        # from config.yaml we can receive a list
+        if isinstance(queues, six.string_types):
+            self.input_queues = [x.strip() for x in queues.split(',')]
+        elif isinstance(queues, list):
+            self.input_queues = queues
+        else:
+            self.input_queues = []
+
+        self.aws_access_key = self._get_config_entry('aws_access_key_id')
+        self.aws_secret_key = self._get_config_entry('aws_secret_access_key')
+        self.aws_region = self._get_config_entry('region')
+
+        self.max_number_of_messages = self._get_config_entry('max_number_of_messages',
+                                                             prefix='sqs_other')
+
+        # checker configuration is update, or not
+        def _is_same_credentials():
+            c = self.session.get_credentials()
+            return c is not None and \
+                c.access_key == self.aws_access_key and \
+                c.secret_key == self.aws_secret_key and \
+                self.session.region_name == self.aws_region
+
+        if self.session is None or not _is_same_credentials():
+            self._setup_sqs()
+
+    def _setup_sqs(self):
+        ''' Setup Boto3 structures '''
+        self._logger.debug('Setting up SQS resources')
+        self.session = Session(aws_access_key_id=self.aws_access_key,
+                               aws_secret_access_key=self.aws_secret_key,
+                               region_name=self.aws_region)
+
+        try:
+            self.sqs_res = self.session.resource('sqs')
+        except NoRegionError:
+            self._logger.warning("The specified region '%s' is invalid", self.aws_region)
+
+    def _get_queue_by_name(self, queueName):
+        ''' Fetch QUEUE by it's name create new one if queue doesn't exist '''
+        try:
+            return self.sqs_res.get_queue_by_name(QueueName=queueName)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'AWS.SimpleQueueService.NonExistentQueue':
+                self._logger.warning("SQS Queue: %s doesn't exist, creating it.", queueName)
+                return self.sqs_res.create_queue(QueueName=queueName)
+            elif e.response['Error']['Code'] == 'InvalidClientTokenId':
+                self._logger.warning("Cloudn't operate sqs because of invalid credential config")
```

Comment:
Should we also throw (and abort sensor processing) on this error?

Reply:
I don't know why this was logged and not raised. It shouldn't be possible to get here if you have invalid credentials. We tested that scenario, and it raises an exception. I can't recall if it did so when getting the resource or creating the session.